crates/client/rpc/src/lib.rs

```diff
@@ -150,7 +153,85 @@ where
         u64::try_from(block.header().block_number).map_err(|e| format!("Failed to convert block number to u64: {e}"))
     }
 
-    /// Helper function to filter Starknet events provided a RPC event filter
+    /// Helper function to get Starknet block details
+    ///
+    /// # Arguments
+    ///
+    /// * `block_id` - The Starknet block id
+    ///
+    /// # Returns
+    ///
+    /// * `(block_events: Vec<EventWrapper>, block: Block)` - A tuple of the block events in
+    ///   block_id and an instance of Block
+    fn get_block_events(&self, block_id: u64) -> Result<(Vec<EventWrapper>, Block), StarknetRpcApiError> {
+        let substrate_block_hash =
+            self.substrate_block_hash_from_starknet_block(BlockId::Number(block_id)).map_err(|e| {
+                error!("'{e}'");
+                StarknetRpcApiError::BlockNotFound
+            })?;
+
+        let block = get_block_by_block_hash(self.client.as_ref(), substrate_block_hash).ok_or_else(|| {
+            error!("Failed to retrieve block");
+            StarknetRpcApiError::BlockNotFound
+        })?;
+        let block_events = self
+            .overrides
+            .for_block_hash(self.client.as_ref(), substrate_block_hash)
+            .events(substrate_block_hash)
+            .unwrap_or_else(|| {
+                info!("No events found in block {}", block_id);
+                Vec::new()
+            });
+        Ok((block_events, block))
+    }
+
+    /// Helper function to get filter events using address and keys
+    ///
+    /// # Arguments
+    ///
+    /// * `events` - A vector of all events
+    /// * `address` - Address to use to filter the events
+    /// * `keys` - Keys to use to filter the events. An event is filtered if any key is present
+    /// * `max_results` - Optional, indicated the max events that need to be filtered
+    ///
+    /// # Returns
+    ///
+    /// * `(block_events: Vec<EventWrapper>, continuation_token: usize)` - A tuple of the filtered
+    ///   events and the first index which still hasn't been processed block_id and an instance of
+    ///   Block
+    pub fn filter_events_by_params(
```

Comment:
This does not need to be part of a `impl` block.
It does not use `self` at all. It can just be a regular function.
This way you can test it without building the `Starknet` type.

And you can test it here, in `client/rpc`, rather than in `node`

Reply:
moved it outside the impl block and move the test cases within rpc